packages/plugins/accounts-manager/src/index.ts

```diff
@@ -73,84 +34,27 @@ export default class AccountsManager {
-    if (!this.ready) {
-      return callback(null, params);
-    }
-    if (params.reqData.method === blockchainConstants.transactionMethods.eth_sendTransaction && this.accounts.length) {
-      // Check if we have that account in our wallet
-      const account = this.accounts.find((acc) => Web3.utils.toChecksumAddress(acc.address) === Web3.utils.toChecksumAddress(params.reqData.params[0].from));
-      if (account && account.privateKey) {
-        return this.signTransactionQueue.push({ payload: params.reqData.params[0], account }, (err: any, newPayload: any) => {
-          if (err) {
-            return callback(err, null);
-          }
-          params.reqData.method = blockchainConstants.transactionMethods.eth_sendRawTransaction;
-          params.reqData.params = [newPayload];
-          callback(err, params);
-        });
-      }
-    }
-    callback(null, params);
-  }
-
-  private async checkBlockchainResponse(params: any, callback: (error: any, result: any) => void) {
-    if (!this.ready) {
-      return callback(null, params);
-    }
-    if ((params.reqData.method === blockchainConstants.transactionMethods.eth_accounts ||
-      params.reqData.method === blockchainConstants.transactionMethods.personal_listAccounts) && this.accounts.length) {
-      if (!arrayEqual(params.respData.result, this.nodeAccounts)) {
-        this.nodeAccounts = params.respData.result;
+  get accounts() {
+    return (async () => {
+      if (!this._accounts) {
         const web3 = await this.web3;
-        this.accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, this.nodeAccounts);
+        this._accounts = await web3.eth.getAccounts();
       }
-      params.respData.result = this.accounts.map((acc) => {
-        if (acc.address) {
-          return acc.address;
-        }
-        return acc;
-      });
-      return callback(null, params);
-    }
-    callback(null, params);
+      return this._accounts || [];
+    })();
   }
 
-  private async parseAndFundAccounts(accounts: any[] | null) {
+  private async parseAndFundAccounts() {
     const web3 = await this.web3;
+    const accounts = await this.accounts;
 
-    const nodeAccounts = await web3.eth.getAccounts();
-    this.nodeAccounts = nodeAccounts;
-    this.accounts = AccountParser.parseAccountsConfig(accounts || this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, nodeAccounts);
-
-    if (!this.accounts.length || !this.embark.config.blockchainConfig.isDev) {
-      this.ready = true;
+    if (!accounts.length || !this.embark.config.blockchainConfig.isDev) {
       return;
     }
     try {
       const coinbase = await web3.eth.getCoinbase();
-      const accts = this.accounts
-        .filter((account) => account.address);
+      const accts = accounts
+        .filter((account) => account && account.address);
```

Comment:
This is the line that makes things blow up right now with

```
Error returned from the node: Returned error: insufficient funds for gas * price + value
```

After some debugging I found out that the `account` used to deploy the contracts has zero funds. The reason for that is that in this line here, we're filtering out all accounts that don't have an `address`.

Unfortunately, when this code is run, none of the accounts will have an `address`, rather, they are addresses themselves. The reason for that is because of how we changed the retrieval of accounts [here](https://github.com/embark-framework/embark/pull/1850/commits/9bea7121d6e54ed8f079a249690364aef5a9b735#diff-cff683ca704c06d26b59ecd676319e87R41). 

E.g. we're relying purely on `web3.eth.getAccounts()` which will always return a list of strings (addresses), therefore, this filtering here causes ground zero.

This used to work before, when we were relying on `AccountParser.parseAccountsConfig()`, which we now also do [here](https://github.com/embark-framework/embark/pull/1850/commits/9bea7121d6e54ed8f079a249690364aef5a9b735#diff-9c48cf3ec968937697f7de95f2410e83R49).

To make this work, we can simply remove this filter and change `acct.address` to `acct` [here](https://github.com/embark-framework/embark/pull/1850/commits/9bea7121d6e54ed8f079a249690364aef5a9b735#diff-cff683ca704c06d26b59ecd676319e87L156), **however** at this point I'm not sure if we indeed really just want to rely on `web3.eth.getAccounts()` or if we still need/want to consider `parseAccountsConfig()`.

Thoughts @emizzle ?